aquadoggo/src/network/peers/message.rs

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use serde::{Deserialize, Serialize};
+
+use crate::replication::{AnnouncementMessage, SyncMessage};
+
+/// p2panda protocol messages which can be sent over the wire.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum PeerMessage {
+    /// Announcement of peers about the schema ids they are interest in.
+    Announce(AnnouncementMessage),
+
+    /// Replication status and data exchange.
+    SyncMessage(SyncMessage),
+}
```

Comment:
We might need a specification change here because without a message type identifier for `Announce` it will be hard for serde to distinct between these variants.